runners/flink/src/test/java/org/apache/beam/runners/flink/streaming/UnboundedSourceWrapperTest.java

```diff
@@ -187,10 +188,9 @@ public void collect(
                 @Override
                 public void close() {}
               });
-        } catch (SuccessException e) {
+        } finally {
           processingTimeUpdateThread.interrupt();
```

Comment:
In a low probability case when the thread was not sleeping at the time of interrupt, the interrupted exception will not be thrown. 
Instead of interrupting shall we use an atomic boolean etc to terminate the while loop?
Alternatively we can also check on Thread.currentThread().isInterrupted() .